doc/developer-documentation.md

```diff
@@ -69,13 +71,15 @@ jobs:
       tree:
         - <tree-name>
         - !<tree-name2>
+        - <tree-name3:branch-name>
 ```
 Here is the description of each field:
 - **`template`**: A `jinja2` template should be added to the [`config/runtime`](https://github.com/kernelci/kernelci-pipeline/tree/main/config/runtime) directory. This template will be used to generate the test definition.
 - **`kind`**: The `kind` field specifies the type of job. It should be `kbuild` for build jobs, `job` for a test suite, and `test` for a single test case.
 - **`image`**: The `image` field specifies the Docker image used for building and running the test. This field is optional. For example, LAVA test jobs use an image defined in the test definition template instead.
 - **`params`**: The `params` field includes parameters for building the kernel (for `kbuild` jobs) or running the test. These parameters can include architecture, compiler, defconfig options, job timeout, etc.
-- **`rules`**: The `rules` field defines job rules. If a test should be scheduled for a specific kernel tree, branch, or version, these rules can be specified here. The rules prefixed with `!` exclude the specified condition from job scheduling. For example, in the given scenario, the scheduler does not schedule a job if an event is received for the kernel tree `tree-name2`.
+- **`rules`**: The `rules` field defines job rules. If a test should be scheduled for a specific kernel tree, branch, or version, these rules can be specified here. The rules prefixed with `!` exclude the specified condition from job scheduling. For example, in the given scenario, the scheduler does not schedule a job if an event is received for the kernel tree `tree-name2`. It is also possible to schedule job for a specific branch of a tree. For example, a rule `tree:branch` can be mentioned under
+`tree` or `branch` section to define that.
```

Comment:
```suggestion
`tree` or `branch` section to define that, still accepting the `!` prefix to ensure jobs can be scheduled on all branches from a given tree, except a few select ones.
Rules can be defined in relation to any attribute present in the parent node (such as `defconfig` or `fragments` -- or even `arch` for jobs having a parent node of the `kbuild` type).
`min_version` and `max_version` are special cases allowing to allow jobs to run only on given kernel versions.
```

This would probably deserve a (sub)paragraph of its own though, so feel free to disregard this suggestion and I'll follow up on this asap.

Reply:
I'll add these lines.